src/test_l1_dsFPD.c

```diff
@@ -1447,11 +1426,11 @@ void test_l1_dsFPD_positive_dsGetFPColor (void)
     // Step 02: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState() 
     // Step 03: Call dsGetFPColor() with all valid indicators
     // Step 04: Call dsGetFPColor() again and compare results
-    count = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+    count = UT_KVP_PROFILE_GET_UINT8("dsFPD/Number_of_Indicators");
     for (int i = 1; i <= count; i++)
     {
         snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
-        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
```

Comment:
At line 1442, can we also check that the retrieved colour is also part of test profile to validate.

Reply:
Done

Reply:
Should you use the following steps to validate?
for (int k = 1; k < numOfSupportedColors; ++k)
            {
                snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors/%d", eIndicator, j);
                color =(dsFPDColor_t)UT_KVP_PROFILE_GET_UINT32(buffer);
                if (retrievedColor == color)
                {
                    isSupported = 1;
                     break;
                }

            }
            if (!isSupported)
            {
               UT_ASSERT_EQUAL(retrievedColor, color);
            }

Reply:
Done